build-packages/scandipwa-extensibility/build-config/babel-plugin-middleware-decorator/index.js

```diff
@@ -71,9 +71,59 @@ const addSuperToConstructor = (path, types) => {
     constructor.get('body').unshiftContainer('body', superCall);
 };
 
+/**
+ * Attempt to read package.json located by p
+ * @param {string} packageJsonPath 
+ */
+const getPackageJson = (packagePath) => {
+    const packageJsonPath = path.join(packagePath, 'package.json');
+
+    try {
+        return require(packageJsonPath);
+    } catch (err) {
+        return {}
+    }
+}
+
+/**
+ * Memoization strategy is valid only for functions with 1 argument
+ * This strategy works better with 1-argument functions
+ * 
+ * @param {function} cb 
+ */
+const memoizeBySingleArgument = (cb) => {
+    const memoizedValues = {};
+
+    return (arg) => {
+        // Handle new value
+        if (!memoizedValues[arg]) {
+            const result = cb(arg);
+            memoizedValues[arg] = result;
+        }
+        
+        return memoizedValues[arg];
+    };
+}
+
+const isScandipwaPackageFile = memoizeBySingleArgument((filePath) => {
+    // Potentially, only the files inside of the src/ directory need to be transpiled
+    const src = filePath.lastIndexOf('/src/');
+
+    const packagePath = filePath.slice(0, src);
+    const packageJson = getPackageJson(packagePath);
+
+    // If package.json has `scandipwa` configurations => is a scandipwa package
+    if (packageJson.scandipwa) {
+        return true;
+    }
```

Comment:
This code does not look safe to me, what if the `src` last index is -1 => it will slide a last letter from the string, then it will try to get package.json from it. Yes, it will return `{}` but I would prefer not to try resolving the package in the first place.

Reply:
Fixed. No src => not a ScandiPWA package.